src/panels/history/ha-panel-history.ts

```diff
@@ -315,7 +315,7 @@ class HaPanelHistory extends LitElement {
       this._startDate,
       this._endDate,
       statisticIds,
-      "hour",
+      differenceInDays(this._endDate, this._startDate) > 7 ? "day" : "hour",
```

Comment:
This is pretty big change, so if you select 8 days, you will only get 8 datapoints?

Reply:
Only for statistics. You'll still get detailed history data. We can change the exact number of days but there should be a limit somewhere

Reply:
Can we move this change out of a patch and do it in the beta?